bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BStruct.java

```diff
@@ -173,6 +180,188 @@ public void setRefField(int index, BRefType value) {
         refFields[index] = value;
     }
 
+    @Override
```

Comment:
The cleanest approach is to introduce another concrete implementation of the StructureType interface for the global memory block. Introduce another interface say LockableStructureType and add all the lock and unlock methods to that interface.

This will cleanup the BStruct and BConnector classes.

Reply:
done